### PR TITLE
Set jsonc for c_cpp_properties.json.

### PR DESCRIPTION
**Pull request by sean-mcmanus**
Fix c_cpp_properties.json not opening as JSON with comments. Follow up to https://github.com/microsoft/vscode-cpptools/issues/5885
Fix the configuration UI disappearing when selection moves to the C/C++ or Log Diagnostics window, mentioned at https://github.com/microsoft/vscode-cpptools/issues/5822 .
Fix an unreported bug where the configuration UI wouldn't appear for non-cpp files that are active on startup (until the selection changed).

### CODE REVIEW

File: Extension/src/LanguageServer/extension.ts [hunk 1]
```diff
@@ -452,6 +452,7 @@ function realActivation(): void {
 
     disposables.push(vscode.workspace.onDidChangeConfiguration(onDidChangeSettings));
     disposables.push(vscode.window.onDidChangeActiveTextEditor(onDidChangeActiveTextEditor));
+    ui.activeDocumentChanged(); // Handle already active documents (for non-cpp files that we don't register didOpen).
```
Comment (sean-mcmanus) [inline Extension/src/LanguageServer/extension.ts:455]: This was also causing a bug where the config wasn't being shown initially for open non-cpp documents until a cursor change was done.

File: Extension/src/LanguageServer/ui.ts [hunk 2]
```diff
@@ -164,9 +164,13 @@ export class UI {
...
             // It's sometimes desirable to see the config and icons when making settings changes.
-            const isSettingsJson: boolean = ((activeEditor.document.fileName.endsWith("c_cpp_properties.json") || activeEditor.document.fileName.endsWith("settings.json")));
-
-            this.ShowConfiguration = isCpp || isSettingsJson;
+            const isCppPropertiesJson: boolean = activeEditor.document.fileName.endsWith("c_cpp_properties.json");
+            if (isCppPropertiesJson) {
+                vscode.languages.setTextDocumentLanguage(activeEditor.document, "jsonc");
+            }
+            this.ShowConfiguration = isCpp || isCppPropertiesJson ||
+                activeEditor.document.uri.scheme === "output" ||
```
Comment (sean-mcmanus) [inline Extension/src/LanguageServer/ui.ts:174]: This is an unrelated change -- so the configuration doesn't disappear when in C/C++ or Log Diagnostics output.

**Review by Colengms — APPROVED** [review]
